Document `CALL { ... } IN TRANSACTIONS` (#1271)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/general/query-transactions.asciidoc

This section describes how Cypher queries work with database transactions.
--

All Cypher queries run within transactions.
Modifications done by updating queries are held in memory by the transaction until it is committed, at which point the changes are persisted to disk and become visible to other transactions.
If an error occurs - either during query evaluation, such as division by zero, or during commit, such as constraint violations - the transaction is automatically rolled back, and no changes are persisted in the graph.

> In short, updating queries will always either fully succeed, or not succeed at all.

Reviewer: ```suggestion
In short, updating a query always either fully succeeds, or does not succeed at all.
```

> Note that, as the transaction holds changes in memory, a query that makes a large number of updates will consequently use large amounts of memory.

Reviewer: ```suggestion
[NOTE]
====
A query that makes a large number of updates consequently uses large amounts of memory since the transaction holds changes in memory.
For memory configuration in Neo4j, see the <<operations-manual#memory-configuration, Neo4j Operations Manual -> Memory configuration>>.
====
```

> For memory configuration in Neo4j, see the <<operations-manual#memory-configuration, Neo4j Operations Manual -> Memory configuration>>.

Reviewer: ```suggestion

```

This section describes how Cypher queries work with database transactions.
--

> All Cypher queries run within transactions. Modifications done by updating queries are held in memory by the transaction until it is committed, at which point the changes are persisted to disk and become visible to other transactions.

Reviewer 2: Missing newline after first sentence.

Author: It will all be one contiguous paragraph anyway won't it?

Reviewer 2: True, but I think that for markup files it's good practice to have each sentence on a new line. It's easier to edit and `diffs` will be a lot easier to look at. This resource even quotes the writing guide for `Neo4j Documentation` (which I didn't know existed): https://asciidoctor.org/docs/asciidoc-recommended-practices/

If an error occurs - either during query evaluation, such as division by zero, or during commit, such as constraint violations - the transaction is automatically rolled back, and no changes are persisted in the graph.

In short, updating queries will always either fully succeed, or not succeed at all.

Note that, as the transaction holds changes in memory, a query that makes a large number of updates will consequently use large amounts of memory.
For memory configuration in Neo4j, see the <<operations-manual#memory-configuration, Neo4j Operations Manual -> Memory configuration>>.

Transactions can be either _explicit_ or _implicit_.

- _Explicit_ transactions,
  * are opened by the user,
  * can execute multiple Cypher queries in sequence,

>   * are committed, or rolled back, by the user

Reviewer 2: I think this might look better, but just my 2 cents. 

```suggestion
- _Explicit_ transactions:
  * are opened by the user.
  * can execute multiple Cypher queries in sequence.
  * are committed, or rolled back, by the user.
```

- _Implicit_ transactions, sometimes called auto-commit transactions or auto transactions,
  * are opened automatically,
  * can execute a single Cypher query,

>   * are committed automatically when the query finishes successfully

Reviewer 2: ```suggestion
- _Implicit_ transactions, sometimes called auto-commit transactions or auto transactions:
  * are opened automatically.
  * can execute a single Cypher query.
  * are committed automatically when the query finishes successfully.
```
